infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/configurator/GitconfigConfigurator.java

```diff
@@ -76,24 +76,21 @@ public void configure(NamespaceResolutionContext namespaceResolutionContext, Str
     KubernetesClient client = cheServerKubernetesClientFactory.create();
     Optional<String> gitconfigOptional = getGitconfig(client, namespaceName);
     Optional<Pair<String, String>> usernameAndEmailFromGitconfigOptional = Optional.empty();
-    Optional<Pair<String, String>> usernameAndEmailFromFetcherOptional =
-        getUsernameAndEmailFromFetcher(namespaceName);
     if (gitconfigOptional.isPresent()) {
       String gitconfig = gitconfigOptional.get();
       usernameAndEmailFromGitconfigOptional = getUsernameAndEmailFromGitconfig(gitconfig);
     }
-    if (needUpdateGitconfigConfigmap(
-        usernameAndEmailFromGitconfigOptional, usernameAndEmailFromFetcherOptional)) {
-      ConfigMap gitconfigConfigmap = buildGitconfigConfigmap();
-      Optional<Pair<String, String>> usernameAndEmailOptional =
-          usernameAndEmailFromGitconfigOptional.isPresent()
-              ? usernameAndEmailFromGitconfigOptional
-              : usernameAndEmailFromFetcherOptional;
-      Optional<String> gitconfigSectionsOptional =
-          generateGitconfigSections(gitconfigOptional, usernameAndEmailOptional);
-      gitconfigConfigmap.setData(
-          ImmutableMap.of(CONFIGMAP_DATA_KEY, gitconfigSectionsOptional.orElse("")));
-      client.configMaps().inNamespace(namespaceName).createOrReplace(gitconfigConfigmap);
+    if (usernameAndEmailFromGitconfigOptional.isEmpty()) {
+      Optional<Pair<String, String>> usernameAndEmailFromFetcher =
+          getUsernameAndEmailFromFetcher(namespaceName);
+      if (usernameAndEmailFromFetcher.isPresent()) {
+        ConfigMap gitconfigConfigmap = buildGitconfigConfigmap();
+        Optional<String> gitconfigSectionsOptional =
+            generateGitconfigSections(gitconfigOptional, usernameAndEmailFromFetcher);
+        gitconfigConfigmap.setData(
+            ImmutableMap.of(CONFIGMAP_DATA_KEY, gitconfigSectionsOptional.orElse("")));
+        client.configMaps().inNamespace(namespaceName).createOrReplace(gitconfigConfigmap);
+      }
```

Comment:
can we extract it to a separate method and add clear javadoc?

Reply:
Added javadocs to each step.